src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,50 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService {
+    private BulldozerProducer bulldozerProducer = new BulldozerProducer();
+    private ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+    private TruckProducer truckProducer = new TruckProducer();
+
+    @Override
+    public List getAll(Class type) {
+        if (type == Bulldozer.class) {
+            List<Bulldozer> bulldozers = bulldozerProducer.get();
+            return new ArrayList<>(bulldozers);
+        }
+        if (type == Excavator.class) {
+            List<Excavator> excavators = excavatorProducer.get();
+            return new ArrayList<>(excavators);
+        }
+        if (type == Truck.class) {
+            List<Truck> trucks = truckProducer.get();
+            return new ArrayList<>(trucks);
+        }
+        return new ArrayList();
+    }
+
+    @Override
+    public void startWorking(List machines) {
```

Comment:
```suggestion
    public void startWorking(List<? extends Machine> machines) {
```

Reply:
Why should we parametrize this method again in implementation? @aleksandr-bevza